src/Driver/PDO/Statement.php

```diff
@@ -117,12 +133,109 @@ public function execute($params = null): ResultInterface
             );
         }
 
+        $resourceOffsets = $this->getResourceOffsets();
         try {
             $this->stmt->execute($params);
         } catch (PDOException $exception) {
             throw Exception::new($exception);
+        } finally {
+            if ($resourceOffsets !== null) {
+                $this->restoreResourceOffsets($resourceOffsets);
+            }
         }
 
         return new Result($this->stmt);
     }
+
+    /**
+     * Track a binary parameter reference at binding time. These
+     * are cached for later analysis by the getResourceOffsets.
+     *
+     * @param mixed $resource
+     */
+    private function trackParamResource($resource): void
+    {
+        if (! is_resource($resource)) {
+            return;
+        }
+
+        $resources   = &$this->paramResources;
+        $resources ??= [];
+        $resources[] = $resource;
+    }
+
+    /**
+     * Determine the offset that any resource parameters needs to be
+     * restored to after the statement is executed. Call immediately
+     * before execute (not during bindValue) to get the most accurate offset.
+     *
+     * @return int[]|null Return offsets to restore if needed.
+     */
+    private function getResourceOffsets()
+    {
+        $resources = &$this->paramResources;
```

Comment:
Again, do we need this pointer? If not, please operate on the property directly.

Reply:
I guess the question is again the cost of establishing the local reference. I treat a $this->expr as a dereference and put them in a local instead of repeating the deref (an extra 2+n times in this case).

Reply:
No, sorry, that pointer just makes your code more obscure. Directly working with the property is just fine.